base_layer/core/src/consts.rs

```diff
@@ -40,3 +41,14 @@ pub const MEMPOOL_PENDING_POOL_STORAGE_CAPACITY: usize = 1000;
 pub const MEMPOOL_REORG_POOL_STORAGE_CAPACITY: usize = 1000;
 /// The time-to-live duration used for transactions stored in the ReorgPool
 pub const MEMPOOL_REORG_POOL_CACHE_TTL: Duration = Duration::from_secs(300);
+
+thread_local! {
```

Comment:
Isn't this just re-implementing `thread_rng()`?

Reply:
It does look like it is similar to thread_rng. This is how the the rng in the comms layer was created.